### PR TITLE
Longest prefix matching

### PR DESCRIPTION
**Pull request by jteo1**
Added longest prefix matching functionality for finding handlers based on URIs, and also added a logging flag -s for complete suppression of logging (mostly used for the integration_test.sh to prevent clutter). Integration test also updated to include a test for longest prefix matching.

### CODE REVIEW

File: config [hunk 1]
```diff
@@ -4,11 +4,15 @@ port 8080;
...
+path /static/same StaticHandler {
+	root static;
+}
 path /static StaticHandler {
-	root static/;
+	root static;
 }
 path /special StaticHandler {
-	root static/more/;
+	root static/more;
```
Comment (dstephan1) [inline config]: Can you explain this?

Comment (kevinazhu) [inline config]: I was messing around with it when debugging and happened to change it. It doesn't seem to have any effect, but do you have a preference?

Comment (kevinazhu) [inline config]: Also the given API doesn't seem to have a slash at the end either

Comment (dstephan1) [inline config]: It doesn't matter. I was just wondering if it made a difference. Everything else looks good.

**Pull request merged**